### PR TITLE
Setup ruff, configure it and fix some problems

### PR DESCRIPTION
**Pull request by razzeee**
Replace pylint, isort and flake. 

Ruff should cover all the above and more. It's used by fastapi and other big projects, so it seems safe enough.

### CODE REVIEW

File: backend/pyproject.toml [hunk 1]
```diff
@@ -44,5 +42,7 @@ httpx = "^0.23.3"
 requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
-[tool.isort]
-profile = "black"
+[tool.ruff]
+ignore = ["E501"]
+exclude = ["alembic/versions/*"]
+extend-select = ["I", "SIM", "W", "UP", "TCH"]
```
Comment (razzeee) [inline backend/pyproject.toml:48]: I = isort
F is on by default and stands for flake see https://beta.ruff.rs/docs/rules/#pyflakes-f

**Comment by razzeee** [comment]
Most of the code changes have been automatic fixes, but I have reviewed all and they at least seem to make sense

File: backend/pyproject.toml [hunk 2]
```diff
@@ -44,5 +42,7 @@ httpx = "^0.23.3"
 requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
-[tool.isort]
-profile = "black"
+[tool.ruff]
+ignore = ["E501"]
```
Comment (razzeee) [inline backend/pyproject.toml:46]: We're ignoring the line too long error. We did not enforce this before and it would cause some problems.

**Comment by barthalion** [comment]
the lock file needs regeneration

**Comment by razzeee** [comment]
Rebased/regenerated

**Pull request merged**